setuptools/tests/test_egg_info.py

```diff
@@ -486,6 +486,57 @@ def test_doesnt_provides_extra(self, tmpdir_cwd, env):
             pkg_info_text = pkginfo_file.read()
         assert 'Provides-Extra:' not in pkg_info_text
 
+    @pytest.mark.parametrize("files, license_in_sources", [
+        ({
+            'setup.cfg': DALS("""
+                              [metadata]
+                              license_file = LICENSE
+                              """),
+            'LICENSE': DALS("Test license")
+        }, True), # with license
+        ({
+            'setup.cfg': DALS("""
+                              [metadata]
+                              license_file = INVALID_LICENSE
+                              """),
+            'LICENSE': DALS("Test license")
+        }, False), # with an invalid license
+        ({
+            'setup.cfg': DALS("""
+                              """),
+            'LICENSE': DALS("Test license")
+        }, False), # no license_file attribute
+        ({
+            'setup.cfg': DALS("""
+                              [metadata]
+                              license_file = LICENSE
+                              """),
+            'MANIFEST.in': DALS("exclude LICENSE"),
+            'LICENSE': DALS("Test license")
+        }, False) # license file is manually excluded
+    ])
+    def test_setup_cfg_license_file(
+            self, tmpdir_cwd, env, files, license_in_sources):
+        self._create_project()
+        build_files(files)
+        environ = os.environ.copy().update(
```

Comment:
`dict.update` mutates an object in place and returns `None`, so this is a no-op.

`os.environ.copy()` creates a copy of `os.environ`, then `update` mutates that copy and returns `None`. `environ` is set to `None` and the ephemeral copy that was mutated is deleted since nothing holds a reference to it. I would do it this way:

```python
environ = os.environ.copy()
environ['HOME'] = env.paths['home']
```

However, I don't understand what this line is intended to do. Is it actually necessary, considering the tests seem to be passing fine without it?

Reply:
Removed these statements from this test.

I originally added them as this block looked to be a common pattern in this test component (e.g. `test_python_requires_egg_info` or `test_long_description_content_type` etc). I copied it over as I thought that not having this may have unintended side effects that I couldn't spot.